zebra/rt_netlink.c

```diff
@@ -1290,65 +1201,37 @@ static int netlink_route_change_read_unicast_internal(struct nlmsghdr *h,
 				}
 			}
 		}
-		if (nhe_id || ng) {
-			rib_add_multipath(afi, SAFI_UNICAST, &p,
-					  (struct prefix_ipv6 *)&src_p,
-					  re, ng, startup);
-			if (ng)
-				nexthop_group_delete(&ng);
-		} else {
-			/*
-			 * I really don't see how this is possible
-			 * but since we are testing for it let's
-			 * let the end user know why the route
-			 * that was just received was swallowed
-			 * up and forgotten
-			 */
-			zlog_err(
-				"%s: %pFX multipath RTM_NEWROUTE has a invalid nexthop group from the kernel",
-				__func__, &p);
-			zebra_rib_route_entry_free(re);
+		dplane_ctx_route_set_is_multipath(ctx, is_multipath);
+		dplane_ctx_route_set_nexthop_group(ctx, ng);
+
+		if (ng) {
```

Comment:
can we find a way to have the context object take over the list of nexthops, instead of having to allocate and free two copies?